### PR TITLE
make only the operator open/close

### PR DESCRIPTION
**Pull request by wmui51**

### CODE REVIEW

File: packages/tree-view/src/behaviors/TreeItemBehavior.js [hunk 1]
```diff
@@ -69,16 +67,28 @@ export default class TreeItemBehavior extends Component {
...
+    }
+  };
+
+  handleOperatorClick = (event, treeItem) => {
+    event.stopPropagation();
+
+    if (this.props.payload) {
+      // eslint-disable-next-line no-param-reassign
+      treeItem.meta.collapsed = !treeItem.meta.collapsed;
+      this.setIsCollapsed(treeItem.meta.collapsed);
```
Comment (wmui51) [inline packages/tree-view/src/behaviors/TreeItemBehavior.js]: this is the line that sets collapsed... the `setIsCollapsed` doesn't seem to do anything.

Comment (rroxas) [reply packages/tree-view/src/behaviors/TreeItemBehavior.js]: it does, without that line the open/close wont work properly for the tree object.

**Comment by wmui51** [comment]
@rroxas checkout this commit https://github.com/wmui51/hig/pull/3/commits/f0f6e5249a7bf2869b2ce26f55d71bab638e2177

animation seems to be working, but now it seems like the keyboard isn't updating the dom nodes, i've consoled a few things out.

**Comment by wmui51** [comment]
i think i've fixed the keyboard stuff too, check last commit to see if there might be a better way.

**Comment by wmui51** [comment]
also if this works, can you test out controlled (`collapsed` set) and `defaultCollapsed` is set. I noticed it hasn't been added to this area: https://github.com/wmui51/hig/pull/3/commits/ad9d665b6391f8f1cb206354fb7e821d71235943 as currently it is only set in the non-object tree.

File: packages/tree-view/src/behaviors/TreeViewBehavior.js [hunk 2]
```diff
@@ -116,7 +116,7 @@ export default class TreeViewBehaviorRR extends Component {
 
     const domNodeList = this.treeViewRef.querySelectorAll("li");
     const treeItemArrayControl =
-      this.getTreeItemArray().length !== domNodeList.length
+      this.getTreeItemArray().length !== domNodeList.length || this.props.treeNode
```
Comment (wmui51) [inline packages/tree-view/src/behaviors/TreeViewBehavior.js:119]: check here for keyboard

**Review by rroxas — APPROVED** [review]

**Pull request merged**